pkg/sql/catalog/schemaexpr/expr.go

```diff
@@ -559,14 +559,17 @@ func ValidateTTLExpirationExpression(
 		)
 	}
 
+	// ttl_expiration_expression used to conservatively require
+	// volatility.Immutable, but now only requires volatility.Stable to support
+	// one of its main use cases: `timestamptz + interval`.
```

Comment:
could you mention the detail about how the session data in a job cannot be changed, so we don't need to worry about non-determinism from that?

though technically, a stable expression can be non-deterministic for reasons not related to the session data. i wonder if this is OK

Reply:
I will add some more details to the comment. I think it is ok if the user references a function where the output changes in between SELECT or DELETE statements when they change a config because it will get picked up the next time the job runs.

Reply:
Or the expression could reference a stable UDF which query some tables in the future. But this is super rare. It worth return a hint if the expression is stable, but your call.